Src/EB/AMReX_EB_utils.cpp

```diff
@@ -196,11 +196,25 @@ facets_nearest_pt (IntVect const& ind_pt, IntVect const& ind_loop, RealVect cons
 }
 }
 
-void FillSignedDistance (MultiFab& mf, EB2::Level const& ls_lev,
-                         EBFArrayBoxFactory const& eb_factory, int refratio,
+void FillSignedDistance (MultiFab& mf_in, EB2::Level const& ls_lev,
+                         EBFArrayBoxFactory const& eb_factory_in, int refratio,
                          bool fluid_has_positive_sign)
 {
-    AMREX_ALWAYS_ASSERT(mf.is_nodal());
+    AMREX_ALWAYS_ASSERT(mf_in.is_nodal());
+
+    // because the algorithm below is N^2 in the number of points per box,
+    // we always do this operation with a max grid size of 32.
+    auto new_ba = mf_in.boxArray();
+    new_ba.convert(IntVect::TheZeroVector());
+    new_ba.maxSize(32);
+    new_ba.convert(IntVect::TheUnitVector());
+    int max_guard = eb_factory_in.getBndryCent().nGrow();
+    auto eb_factory_ptr = amrex::makeEBFabFactory(ls_lev.Geom(), new_ba,
+                                                  DistributionMapping(new_ba),
+                                                  {max_guard, max_guard, max_guard},
+                                                  amrex::EBSupport::full);
+    EBFArrayBoxFactory const& eb_factory = *eb_factory_ptr;
+    MultiFab mf(new_ba, DistributionMapping(new_ba), mf_in.nComp(), mf_in.nGrow());
```

Comment:
The DistributionMapping created here might be different from the DistributionMapping created in the call to maekEBFabFactory. So this is a bug.